helm/hpcc/templates/_helpers.tpl

```diff
@@ -101,36 +118,163 @@ Add ConfigMap volume for a component
 
 {{/*
 Add data volume mount
+If any storage planes are defined that name pvcs they will be mounted
 */}}
 {{- define "hpcc.addDataVolumeMount" -}}
+{{- /*Create local variables which always exist to avoid having to check if intermediate key values exist*/ -}}
+{{- $storage := (.root.Values.storage | default dict) -}}
+{{- $planes := ($storage.planes | default list) -}}
+{{- $dataStorage := ($storage.dataStorage | default dict) -}}
+{{- $spillStorage := ($storage.spillStorage | default dict) -}}
+{{- $daliStorage := ($storage.daliStorage | default dict) -}}
+{{- $dllStorage := ($storage.dllStorage | default dict) -}}
+{{- $daliStoragePlane := ($daliStorage.plane | default "") -}}
+{{- $dllStoragePlane := ($dllStorage.plane | default "") -}}
+{{- range $plane := $planes -}}
+ {{- if $plane.pvc -}}
+  {{- if and (ne $plane.name $daliStoragePlane) (ne $plane.name $dllStoragePlane) -}}
+   {{- $num := int ( $plane.numDevices | default 1 ) -}}
+   {{- if le $num 1 }}
+- name: {{ $plane.name }}-pv
+  mountPath: {{ $plane.prefix | quote }}
+   {{- else }}
+    {{- range $elem := until $num }}
+- name: {{ $plane.name }}-pv-many-{{- add $elem 1 }}
+  mountPath: {{ printf "%s/d%d" $plane.prefix (add $elem 1) | quote }}
+    {{- end }}
+   {{- end }}
+  {{- end }}
+ {{- end }}
+{{- end }}
+{{- if (not $dataStorage.plane) }}
 - name: datastorage-pv
   mountPath: "/var/lib/HPCCSystems/hpcc-data"
+{{- end }}
 {{- end -}}
 
 {{/*
 Add data volume
 */}}
 {{- define "hpcc.addDataVolume" -}}
+{{- /*Create local variables which always exist to avoid having to check if intermediate key values exist*/ -}}
+{{- $storage := (.root.Values.storage | default dict) -}}
+{{- $planes := ($storage.planes | default list) -}}
+{{- $dataStorage := ($storage.dataStorage | default dict) -}}
+{{- $spillStorage := ($storage.spillStorage | default dict) -}}
+{{- $daliStorage := ($storage.daliStorage | default dict) -}}
+{{- $dllStorage := ($storage.dllStorage | default dict) -}}
+{{- $daliStoragePlane := ($daliStorage.plane | default "") -}}
+{{- $dllStoragePlane := ($dllStorage.plane | default "") -}}
+{{- range $plane := .root.Values.storage.planes -}}
```

Comment:
Should this be using $planes? If not,  I don't think $planes is used

Reply:
yes it should